examples/Misc/NavigateSearchResultPagesCachingTokens.php

```diff
@@ -0,0 +1,270 @@
+        ];
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $response = $googleAdsServiceClient->search($customerId, $query, $searchOptions);
+        self::cacheNextPageToken($pageTokens, $response->getPage(), 1);
+
+        // Determines the total number of results and prints it.
+        // The total results count does not take into consideration the LIMIT clause of the query
+        // so we need to find the minimal value between the limit and the total results count.
+        $totalNumberOfResults = min(
+            self::RESULTS_LIMIT,
+            $response->getPage()->getResponseObject()->getTotalResultsCount()
+        );
+        printf('Total number of campaigns found: %d.%s', $totalNumberOfResults, PHP_EOL);
+
+        // Determines the total number of pages and prints it.
+        $totalNumberOfPages = ceil($totalNumberOfResults / self::PAGE_SIZE);
+        printf('Total number of pages: %d.%s', $totalNumberOfPages, PHP_EOL);
+        if (!$totalNumberOfPages) {
+            throw new Exception(
+                'Could not find any results.'
+            );
+        }
+
+        $middlePageNumber = ceil($totalNumberOfPages / 2);
+        printf('%s--- 1. Print results of the page #%d:%1$s%1$s', PHP_EOL, $middlePageNumber);
+        self::fetchAndPrintPageResults(
+            $googleAdsServiceClient,
+            $customerId,
+            $query,
+            $searchOptions,
+            $middlePageNumber,
+            $pageTokens
+        );
+
+        printf('%s--- 2. Print results from the last page to the first:%1$s', PHP_EOL);
+        foreach (range($totalNumberOfPages, 1) as $pageNumber) {
+            printf('%s-- Printing results for page #%d:%1$s', PHP_EOL, $pageNumber);
+            self::fetchAndPrintPageResults(
+                $googleAdsServiceClient,
+                $customerId,
+                $query,
+                $searchOptions,
+                $pageNumber,
+                $pageTokens
+            );
+        }
+    }
+
+    // [START navigate_search_result_pages_caching_tokens]
+    /**
+     * Fetches and prints the results of a page of a search using a cache of page tokens.
+     *
+     * @param GoogleAdsServiceClient $googleAdsServiceClient the Google Ads API Service client
+     * @param int $customerId the customer ID
+     * @param string $searchQuery  the search query
+     * @param array $searchOptions the search options
+     * @param int $pageNumber the number of the page to fetch and print results for
+     * @param int[] &$pageTokens the cache of page tokens to use and update
+     */
+    private static function fetchAndPrintPageResults(
+        GoogleAdsServiceClient $googleAdsServiceClient,
+        int $customerId,
+        string $searchQuery,
+        array $searchOptions,
+        int $pageNumber,
+        array &$pageTokens
+    ) {
+        // There is no need to fetch the pages we already know the page tokens for.
```

Comment:
I think the comment doesn't quite 100% match what comes after it.
I'd swap the if clause, using `isset` instead. This will align with the Google recommendation that prefers positive clauses over the negative ones.